src/Microsoft.DotNet.XHarness.CLI/Commands/Android/AndroidTestCommand.cs

```diff
@@ -120,6 +143,36 @@ protected override Task<ExitCode> InvokeInternal(ILogger logger)
             return Task.FromResult(ExitCode.GENERAL_FAILURE);
         }
 
+        private string GetDeviceToUse(ILogger logger, AdbRunner runner, string apkRequiredArchitecture)
+        {
+            var allDevicesAndTheirArchitectures = runner.GetAttachedDevicesAndArchitectures();
+            if (allDevicesAndTheirArchitectures.Count == 1)
```

Comment:
I quite don't understand what is the difference for `1` and `n` devices. We do the same for both - just return the first attached and warn

Reply:
mostly just because if something went wrong, default has a better chance of just working.  If you send -s <non existent device> somehow you timeout.  I guess you're right.